tests/test_pipelines_object_detection.py

```diff
@@ -0,0 +1,227 @@
+# Copyright 2021 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+
+from transformers import AutoFeatureExtractor, AutoModelForObjectDetection, PreTrainedTokenizer, is_vision_available
+from transformers.pipelines import ObjectDetectionPipeline, pipeline
+from transformers.testing_utils import require_timm, require_torch, require_vision, slow
+
+
+if is_vision_available():
+    from PIL import Image
+else:
+
+    class Image:
+        @staticmethod
+        def open(*args, **kwargs):
+            pass
+
+
+@slow
+@require_timm
+@require_vision
+@require_torch
+class ObjectDetectionPipelineTests(unittest.TestCase):
```

Comment:
Thanks for implementing a nice set of tests! Unfortunately, we went through a refactor with @Narsil that will require updating this file to conform to the new testing format.

Reply:
You can take a look at the new file for testing the image classification pipeline to get an idea of what needs to be done: https://github.com/huggingface/transformers/blob/master/tests/test_pipelines_image_classification.py

Please let us know if you'd like us to handle this refactor.

Reply:
awesome! Yeah, I'd like to handle this refactor